### PR TITLE
Add Import

### PR DESCRIPTION
**Pull request by MstrPikachu**
PR for the import use case.
In this PR, the StringStorage interface was made to represent a class that can store strings, because export is about storing a string into a kind of format. FixedFileStorage and PasteBinStorage are StringStorage classes that write to a file and to pastebin respectively.

There is also an ExportModel class to store what information about a plugin is used in the export (in this case version and name).

Import is currently implemented by installing all the plugins and then returning information about the installs. Currently merging the information is a bit scuffed but everything else should be fine

### CODE REVIEW

File: src/main/java/org/hydev/mcpm/client/commands/controllers/ExportPluginsController.java [hunk 1]
```diff
@@ -28,6 +25,6 @@ public ExportPluginsController(ExportPluginsBoundary boundary) {
      */
     public void export(ExportPluginsInput input, Consumer<String> log) {
```
Comment (kchprog) [inline src/main/java/org/hydev/mcpm/client/commands/controllers/ExportPluginsController.java]: Good.

File: src/main/java/org/hydev/mcpm/client/display/presenters/LogExportPresenter.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import org.hydev.mcpm.client.export.ExportPluginsResult;
+
+import java.util.function.Consumer;
+
+/**
+ * ExportPresenter that presents to a log.
+ */
+public class LogExportPresenter implements ExportPresenter {
+
+    private String getColor(ExportPluginsResult exportPluginsResult) {
```
Comment (kchprog) [inline src/main/java/org/hydev/mcpm/client/display/presenters/LogExportPresenter.java]: Missing docstring

File: src/main/java/org/hydev/mcpm/client/display/presenters/LogExportPresenter.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        return switch (exportPluginsResult.state()) {
+            case SUCCESS -> "&a";
+            case FAILED_TO_FETCH_PLUGINS -> "&c";
+        };
+    }
+
+    private String getMessage(ExportPluginsResult exportPluginsResult) {
```
Comment (kchprog) [inline src/main/java/org/hydev/mcpm/client/display/presenters/LogExportPresenter.java]: Missing docstring

File: src/main/java/org/hydev/mcpm/client/display/presenters/LogExportPresenter.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        return switch ((exportPluginsResult.state())) {
+            case SUCCESS -> exportPluginsResult.export();
+            case FAILED_TO_FETCH_PLUGINS -> "Failed to fetch plugins";
+        };
+    }
+
+
+    @Override
```
Comment (kchprog) [inline src/main/java/org/hydev/mcpm/client/display/presenters/LogExportPresenter.java:28]: Missing docstring

File: src/main/java/org/hydev/mcpm/client/export/ImportResult.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
+package org.hydev.mcpm.client.export;
+
+import java.util.Map;
+
+
+/**
+ * Class storing results of the
```
Comment (kchprog) [inline src/main/java/org/hydev/mcpm/client/export/ImportResult.java]: Incomplete docstring

File: src/main/java/org/hydev/mcpm/client/export/ImportResult.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    private final Map<String, Boolean> installResults;
+
+    /**
+     * Result for importing plugins
+     *
+     * @param installResults Result of each install
+     */
+    public ImportResult(Map<String, Boolean> installResults) {
+        this.installResults = installResults;
+        boolean success = true;
```
Comment (kchprog) [inline src/main/java/org/hydev/mcpm/client/export/ImportResult.java]: These are odd variable names. Consider moving them into the for loop and renaming them to refer to explicit conditions instead.

File: src/main/java/org/hydev/mcpm/client/export/PasteBinStorage.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        if (!token.startsWith("http://") && !token.startsWith("https://"))
+        {
+            token = host + token;
+        }
+
+        // Send request
+        return Request.get(token)
+            .addHeader("content-type", "text/plain")
+            .execute().returnContent().asString();
+    }
```
Comment (kchprog) [inline src/main/java/org/hydev/mcpm/client/export/PasteBinStorage.java]: Just curious on how the IOexception is thrown: can Request.get() throw any other error?

**Review by kchprog — COMMENTED** [review]
Pretty cool, I'll approve when you get the rest of the stuff merged. A few small things.

File: src/main/java/org/hydev/mcpm/client/export/ImportResult.java (hunk 5 above)
Comment (MstrPikachu) [reply src/main/java/org/hydev/mcpm/client/export/ImportResult.java]: LOL whoops

File: src/main/java/org/hydev/mcpm/client/display/presenters/LogExportPresenter.java (hunk 4 above)
Comment (MstrPikachu) [reply src/main/java/org/hydev/mcpm/client/display/presenters/LogExportPresenter.java:28]: The docstring is in the interface

**Review by hykilpikonna — APPROVED** [review]

**Pull request merged**